solr/core/src/java/org/apache/solr/search/ValueSourceParser.java

```diff
@@ -758,6 +758,35 @@ public ValueSource parse(FunctionQParser fp) throws SyntaxError {
       }
     });
 
+    addParser("spayload", new ValueSourceParser() {
+      @Override
+      public ValueSource parse(FunctionQParser fp) throws SyntaxError {
+        // spayload(field,value[,default])
+
+        TInfo tinfo = parseTerm(fp); // would have made this parser a new separate class and registered it, but this handy method is private :/
+
+        ValueSource defaultValueSource;
+        if (fp.hasMoreArguments()) {
+          defaultValueSource = fp.parseValueSource();
+        } else {
+          defaultValueSource = new LiteralValueSource("");
+        }
+
+        IndexSchema schema = fp.getReq().getCore().getLatestSchema();
+        final FieldType fieldType = schema.getFieldType(tinfo.field);
+
+        if (fieldType.getTypeName().equals("delimited_payloads_string")) {
```

Comment:
Relying on a field type name is brittle - should rather do the introspection mechanism.   I think PayloadUtils.getPayloadEncoder(fieldType) might do the trick for you returning "string".